tensorflow/python/keras/applications/efficientnet_weight_update_util.py

```diff
@@ -0,0 +1,307 @@
+  
+  Args:
+    path_h5: str, path to output hdf5 file to write weights loaded
+      from ckpt files.
+    path_ckpt: str, path to the ckpt files (e.g. 'efficientnet-b0/model.ckpt')
+      that records efficientnet weights from original repo 
+      https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet
+    keras_model: keras model, built from keras.applications efficientnet
+      functions (e.g. EfficientNetB0)
+    use_ema: Bool, whether to use ExponentialMovingAverage result or not
+  """
+  model_name_keras = keras_model.name
+  model_name_tf = model_name_keras_to_tf(model_name_keras)
+  keras_model.save_weights(path_h5)
+
+  keras_weights = get_h5_names(path_h5)
+  tf_weights = get_tf_names(path_ckpt)
+  blocks_keras = get_keras_blocks(keras_weights)
+  blocks_tf = get_tf_blocks(tf_weights)
+
+  with tf.compat.v1.Session() as sess:
+    saver = tf.compat.v1.train.import_meta_graph(f'{path_ckpt}.meta')
+    saver.restore(sess, path_ckpt)
+    graph = tf.compat.v1.get_default_graph()
+    v_all = tf.compat.v1.global_variables()
+
+    for keras_block, tf_block in zip(blocks_keras, blocks_tf):
+      print(f'working on block {keras_block}, {tf_block}')
+      for keras_name in keras_weights:
+        if keras_block in keras_name:
+          tf_name = keras_name_to_tf_name_block(keras_name, keras_block=keras_block, tf_block=tf_block, use_ema=use_ema, model_name_tf=model_name_tf)
+          for v in v_all:
+            if v.name == tf_name:
+              v_val = sess.run(v)
+              with h5py.File(path_h5, 'a') as f:
+                v_prev = f[keras_name][()]
+                f[keras_name].write_direct(v_val)
+              print(f'writing from: {tf_name}\n  to: {keras_name}')
+              print(f'  average change: {abs(v_prev - v_val).mean()}')
+              v_all.remove(v)
+              break
+          else:
+            raise ValueError(f'{keras_name} has no match in ckpt file')
+
+    for keras_name in keras_weights:
+      if any([x in keras_name for x in ['stem', 'top', 'predictions', 'probs']]):
+        tf_name = keras_name_to_tf_name_stem_top(keras_name, use_ema=use_ema, model_name_tf=model_name_tf)
+        for v in v_all:
+          if v.name == tf_name:
+            v_val = sess.run(v)
+            with h5py.File(path_h5, 'a') as f:
+              v_prev = f[keras_name][()]
+              try:
+                f[keras_name].write_direct(v_val)
+              except:
+                raise ValueError(f'weight in {tf_name} does not ift into {keras_name}')
+            print(f'writing from: {tf_name}\n  to: {keras_name}')
+            print(f'  average change: {abs(v_prev - v_val).mean()}')
+            v_all.remove(v)
+            break
+
+
+def model_name_keras_to_tf(model_name_keras):
+  """Infer model name in both keras and tf implementations"""
+  model_name_tf = model_name_keras.replace('efficientnet', 'efficientnet-')
+  return model_name_tf
+
+
+def get_h5_names(path_h5):
```

Comment:
This function name doesn't properly describe the functionality.

How about get_variable_names_from_h5()?